src/binding.cc

```diff
@@ -65,6 +68,7 @@ void Interruptor::Interrupt() {
   if (!has_interrupted_) {
     isolate_->TerminateExecution();
     has_interrupted_ = true;
+    interrupted.insert(index_);
```

Comment:
This can be called from any thread, so it would be subject to race conditions and would need to be protected by a mutex (and leak memory, because entries are never removed from the set) – I think it would be better not to do this. I’m also not sure that we need to add a feature, since the handler function could already take care of this in JS land through something like

```js
let interrupted = true;
runInterruptible(() => {
  try {
    ... do actual work ...
  } finally {
    interrupted = false;
  }
});
```

If that doesn’t work (I could absolutely be overlooking something), then I think checking the `interrupted_` boolean in the native `RunInterruptible` method itself and passing that along would make sense.

Reply:
Ah, threads, right! Yeah, makes sense. Also you're right, handling it in JS should be possible like in your example. I'll revert this commit